airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -0,0 +1,215 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import gzip
+import io
+import json
+import urllib.parse as urlparse
+import zipfile
+from abc import ABC, abstractmethod
+from typing import IO, Any, Iterable, List, Mapping, MutableMapping, Optional
+from urllib.parse import parse_qs
+
+import pendulum
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class AmplitudeStream(HttpStream, ABC):
+
+    url_base = "https://amplitude.com/api"
+    api_version = 2
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        respose_data = response.json()
+        yield from respose_data.get(self.name, [])
+
+    def path(self, **kwargs) -> str:
+        return f"/{self.api_version}/{self.name}"
+
+
+class Cohorts(AmplitudeStream):
+    primary_key = "id"
+    api_version = 3
```

Comment:
Why several versions are used?

Reply:
I used documentation for Amplitude, and there, different versions are used for different streams:

1. [All Cohorts](https://developers.amplitude.com/docs/behavioral-cohorts-api#listing-all-cohorts)  - version 3
2. [Active User Counts](https://developers.amplitude.com/docs/dashboard-rest-api#active-and-new-user-counts) - version 2
3. [Export Events](https://developers.amplitude.com/docs/export-api#export-api-parameters) - version 2